cfme/tests/middleware/test_middleware_server.py

```diff
@@ -72,10 +74,115 @@ def test_server_details(provider):
         srv_mgmt.validate_properties()
 
 
+def test_server_reload(provider):
+    """Tests server reload operation on UI
+
+    Steps:
+        * Get servers list from UI
+        * Chooses Wildfly server from list
+        * Clicks 'Reload Server' operation
+        * Checks that server status is not running in UI, in DB and in MGMT.
+        * Waits for some time
+        * Checks that server status is running in UI, in DB and in MGMT.
+    """
+    server_list = MiddlewareServer.servers(provider=provider)
+    for _server in server_list:
+        if _server.product != 'Hawkular':
+            server = _server
+            break
+    assert server, 'Wildfly server is not found in servers list'
+    server.reload_server()
+    flash.assert_success_message('Reload initiated for selected server(s)')
+    # enable when HWKINVENT-185 is fixed
+    # _check_server_stopped(server)
+    sleep(10)
+    _check_server_running(server)
+
+
+def test_hawkular_reload(provider):
+    """Tests Hawkular server itself reload operation message on UI
+
+    Steps:
+        * Get servers list from UI
+        * Chooses Hawkular server from list
+        * Clicks 'Reload Server' operation
+        * Checks that notification message is shown
+        * Checks that server status is running in UI, in DB and in MGMT.
+        * Clicks 'Stop Server' operation
+        * Checks that notification message is shown
+        * Checks that server status is running in UI, in DB and in MGMT.
+    """
+    server_list = MiddlewareServer.servers(provider=provider)
+    for _server in server_list:
+        if _server.product == 'Hawkular':
+            server = _server
+            break
+    assert server, 'Hawkular server is not found in servers list'
+    server.reload_server()
+    flash.assert_success_message('Not reloading Hawkular server')
+    _check_server_running(server)
+    server.stop_server()
+    flash.assert_success_message('Not stopping Hawkular server')
+    _check_server_running(server)
+
+
+''' enable when HWKINVENT-185 is fixed
+def test_server_stop(provider):
+    """Tests server reload operation on UI
+
+    Steps:
+        * Get servers list from UI
+        * Chooses Wildfly server from list
+        * Clicks 'Stop Server' operation
+        * Checks that server status is not running in UI, in DB and in MGMT.
+        * Clicks 'Start Server' operation
+        * Waits for some time
+        * Checks that server status is running in UI, in DB and in MGMT.
+    """
+    server_list = MiddlewareServer.servers(provider=provider)
+    for _server in server_list:
+        if _server.product != 'Hawkular':
+            server = _server
+            break
+    assert server, 'Wildfly server is not found in servers list'
+    server.stop_server()
+    flash.assert_success_message('Stop initiated for selected server(s)')
+    _check_server_stopped(server)
+    sleep(10)
```

Comment:
@hhovsepy we cannot say that server will be `stopped` or `started` in 10 seconds. It is all about it's load. We have to set some `MAXIMUM_TIME_OUT` and do frequent check until it reaches `MAXIMUM_TIME_OUT`.